src/main/java/com/epam/movierating/logic/AccountServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+            throw new ServiceException(e);
+        }
+    }
+
+    @Override
+    public List<Account> getPage(int page, int itemsPerPage) throws ServiceException {
+        if (page < MIN_PAGE_VALUE) {
+            throw new ServiceException("Invalid page parameter: " + page);
+        }
+        int firstItemNumber = (page - 1) * itemsPerPage + 1;
+        try (DaoConnectionManager manager = factory.create()) {
+            AccountDao accountDao = manager.createAccountDao();
+            return accountDao.findBatch(itemsPerPage, firstItemNumber);
+        } catch (Exception e) {
+            throw new ServiceException(e);
+        }
+    }
+
+    @Override
+    public void blockUserById(long id) throws ServiceException {
+        setBlockedByRoleAndId(Role.USER, id, true);
+    }
+
+    @Override
+    public void unblockUserById(long id) throws ServiceException {
+        setBlockedByRoleAndId(Role.USER, id, false);
+    }
+
+    @Override
+    public void blockEditorById(long id) throws ServiceException {
+        setBlockedByRoleAndId(Role.EDITOR, id, true);
+    }
+
+    @Override
+    public void unblockEditorById(long id) throws ServiceException {
+        setBlockedByRoleAndId(Role.EDITOR, id, false);
+    }
+
+    private void setBlockedByRoleAndId(Role role, long id, boolean blocked) throws ServiceException {
+        if (id < MIN_ID_VALUE) {
+            throw new ServiceException("Invalid id value: " + id);
+        }
+        try (DaoConnectionManager manager = factory.create()) {
+            AccountDao accountDao = manager.createAccountDao();
+            Optional<Account> found = accountDao.find(id);
+            if (!found.isPresent()) {
+                throw new ServiceException("No account to set blocked!");
+            }
+            Account account = found.get();
+            boolean blockedActual = account.getBlocked();
+            if (blockedActual == blocked) {
+                throw new ServiceException("Blocked value already set!");
+            }
+            Role roleActual = account.getRole();
+            if (role != roleActual) {
+                throw new ServiceException("Unexpected account role in change blocked operation!");
+            }
+            String userName = account.getUserName();
+            String password = account.getPassword();
+            Account changed = new Account(id, userName, password, roleActual, blocked);
+            accountDao.save(changed);
+        } catch (Exception e) {
+            throw new ServiceException(e);
+        }
+    }
+
+    @Override
+    public void promoteUserToEditor(long id) throws ServiceException {
+        switchRoleByAccountId(Role.USER, Role.EDITOR, id);
+    }
+
+    @Override
+    public void demoteEditorToUser(long id) throws ServiceException {
+        switchRoleByAccountId(Role.EDITOR, Role.USER, id);
+    }
+
+    private void switchRoleByAccountId(Role fromRole, Role toRole, long id) throws ServiceException {
+        if (id < MIN_ID_VALUE) {
```

Comment:
I would not use such checks since in our project they don't bring much value, but reduce readability, if you are still want to check use guava library or sililar from java 8 like Preconditions.checkArgument